bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/DPIUtil.java

```diff
@@ -47,7 +47,7 @@ private static enum AutoScaleMethod { AUTO, NEAREST, SMOOTH }
 	private static AutoScaleMethod autoScaleMethod = AutoScaleMethod.NEAREST;
 
 	private static String autoScaleValue;
-	private static boolean useCairoAutoScale = true;
+	private static final boolean useCairoAutoScale = Platform.PLATFORM.equals("gtk");
```

Comment:
How about:
```suggestion
	private static final boolean IS_LINUX = SWT.getPlatform().equals("gtk");
```

My reasoning:
1. The field became a constant --> use uppercase
2. "Cairo" doesn't seem to bring any useful information anymore, it's just an internal detail about using Linux/Gtk --> exchange "cairo" for "linux" in the name
3. `SWT.getPlatform()` is used in other parts of the class but `Platform.PLATFORM` isn't. You can also choose to change the other parts to `Platform.PLATFORM` and use the same convention everywhere (and reduce 1 level of indirection)

Reply:
Thanks for the pointers. I incorporated 1 and 3. I did not implement 2 as this flag still indicates whether Cairo's auto-scaling is used (which is identified by being on Linux as no other scaling mode is currently used there). Changing to `IS_LINUX` removes semantics, as you will only see that on Linux a scale factor of 1 is always returned without knowing about the reasons (which is that Cairo's auto-scaling is used).